### ISSUE TITLE
Bots creating and checking out. Bank shut client account

### ISSUE THREAD

**Issue by opusuno** [woocommerce/woocommerce#30248]
In case users are actually getting created during checkout despite the setting being turned off, please provide us with a HTTP log with relevant requests, as that would help us troubleshoot this. We're happy to reopen this if we get more information.

_Originally posted by @peterfabian in https://github.com/woocommerce/woocommerce/issues/28711#issuecomment-774054808_

Create account unselected. WCv- 5.4.1 / WPv  5.7.2 / All plugins up to date. Gateway is Payeezy by SkyVerge.

Orders getting placed and declined in rapid succession -

~/website.com-ssl_log:304: 69.167.4.132 - - [09/Jul/2021:09:28:26 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:305: 69.167.4.132 - - [09/Jul/2021:09:28:30 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:306: 69.167.4.132 - - [09/Jul/2021:09:28:31 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:307: 69.167.4.132 - - [09/Jul/2021:09:28:33 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:308: 69.167.4.132 - - [09/Jul/2021:09:28:33 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:309: 69.167.4.132 - - [09/Jul/2021:09:28:34 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:310: 69.167.4.132 - - [09/Jul/2021:09:28:34 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:311: 69.167.4.132 - - [09/Jul/2021:09:28:36 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:312: 69.167.4.132 - - [09/Jul/2021:09:28:38 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:314: 69.167.4.132 - - [09/Jul/2021:09:28:39 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:315: 69.167.4.132 - - [09/Jul/2021:09:28:40 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:316: 69.167.4.132 - - [09/Jul/2021:09:28:41 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:317: 69.167.4.132 - - [09/Jul/2021:09:28:41 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:318: 69.167.4.132 - - [09/Jul/2021:09:28:42 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:319: 69.167.4.132 - - [09/Jul/2021:09:28:44 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:320: 69.167.4.132 - - [09/Jul/2021:09:28:44 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:321: 69.167.4.132 - - [09/Jul/2021:09:28:48 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:322: 69.167.4.132 - - [09/Jul/2021:09:28:49 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:323: 69.167.4.132 - - [09/Jul/2021:09:28:50 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:324: 69.167.4.132 - - [09/Jul/2021:09:28:50 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:325: 69.167.4.132 - - [09/Jul/2021:09:28:52 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:326: 69.167.4.132 - - [09/Jul/2021:09:28:54 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:327: 69.167.4.132 - - [09/Jul/2021:09:28:54 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
~/website.com-ssl_log:328: 69.167.4.132 - - [09/Jul/2021:09:28:55 -0400] "POST /?wc-ajax=checkout HTTP/2.0" 200 194 "https://website.com/checkout/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"

**Comment by juliaamosova** [comment]
Hi @opusuno,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience. 

While you wait, I recommend you check out this article we published on the topic about 6 months ago: [Developer Advisory: Combating Spam Order Bots](https://developer.woocommerce.com/2021/01/14/developer-advisory-combating-spam-order-bots/). It is possible we won't be able to do anything on our end. The article suggests some ways to protect your store fro fake orders.

**Comment by Konamiman** [comment]
Hi @opusuno. As Julia says, we have published [a blog post explaining how to combat spam order posts](https://developer.woocommerce.com/2021/01/14/developer-advisory-combating-spam-order-bots/), also in the original issue a user mentions [the WP Armour plugin](https://wordpress.org/plugins/honeypot/) as an effective tool for that purpose. Given that, we're not going to work on this for the time beign; but I'll leave the issue open in case you (or someone else) want to add further comments.

**Comment by opusuno** [comment]
Thanks for reply @Konamiman 

Current setup is as follows if it can help someone out:  WP Armour is installed and I've selected "must be logged in to checkout" in WC which throws that screen up before checkout. I installed Captcha for the login screens and checkout to rebuff the fake accounts. I also added an addition required field at checkout. Throttling declined transactions on the gateway side to 2 declines per IP is also set.

It's not the most convenient checkout process for humans but it seems to have stopped the attacks for now.   

I can't be sure but the bots were somehow able to create fake accounts, put an item in the cart, then spin 1000s of different CC#s using the same IPs. Transactions were declined but the order notes filled up in the 1000s before the gateway(s) shut the client(s) accounts down. All the server logs look 100% legit but totally abnormal in the shear number of accelerated posts .

Further info for your consideration... 

Both sites that were attacked used SkyVerge plugins (each using different gateways). Not sure there's a connection and I have reached out to them as well.
